Document RBAC requirements and fix section title levels (#40086)
openshift/openshift-docs: modules/sbo-rbac-requirements.adoc

// Module included in the following assemblies:
//
// *openshift-docs/applications/connecting_applications_to_services/exposing-binding-data-from-a-service
[id="sbo-rbac-requirements_{context}"]

> = RBAC requirements

Reviewer: why do not we simply use the Section available in the upstream doc: https://redhat-developer.github.io/service-binding-operator/userguide/exposing-binding-data/rbac-requirements.html ?

Author: @pedjak We have used the section from the upstream doc. But in some places, we need to modify the content a bit to adhere to OpenShift/RH writing guidelines and templates that have sections for Introduction/concept parts and some such. HTH.

> To expose the backing service binding data using the {servicebinding-title}, you require certain Role-based access control (RBAC) permissions. You must specify certain verbs under the `rules` field of the `ClusterRole` resource to allow the RBAC persmissions for the backing service resources. Defining these `rules` enables the {servicebinding-title} with permissions to read the binding data of the the backing service resources throughout the cluster.

Reviewer 2: ```suggestion
To expose the backing service binding data using the {servicebinding-title}, you require certain Role-based access control (RBAC) permissions. Specify certain verbs under the `rules` field of the `ClusterRole` resource to allow the RBAC permissions for the backing service resources. When you define these `rules`, you permit/allow the {servicebinding-title} to read the binding data of the the backing service resources throughout the cluster.
```
The last sentence could be tricky grammatically for some users or localization teams. Adding "you" could simplify it a bit but isn't necessary.

Author: @missmesss Good catch! Thanks, this definitely helps :)

> The {servicebinding-title} performs requests against Kubernetes API using a dedicated service account. By default this account has permissions to bind services to workloads, both represented by the following standard Kubernetes objects:

Reviewer 2: ```suggestion
The {servicebinding-title} performs requests against the(?) Kubernetes API using a dedicated service account. By default, this account has permissions to bind services to workloads, both represented by the following standard Kubernetes objects:
```
If articles aren't used with APIs throughout this set of docs, then ignore my suggestion, please.

* `Deployments`
* `DaemonSets`
* `ReplicaSets`
* `StatefulSets`

> * `DeploymentConfigs`

Author: @pedjak Thanks, mentioning it in the revised version.

> The Operator service account is bound to an aggregated cluster role, allowing Operator providers or cluster administrators to enable binding custom service resources to workloads. You can enclose the required permissions within a `ClusterRole` by labeling it with the `servicebinding.io/controller` flag and setting the flag value to `true`.

Reviewer 2: ```suggestion
The Operator service account is bound to an aggregated cluster role, allowing Operator providers or cluster administrators to enable binding custom service resources to workloads. To provide/grant the required permissions within a `ClusterRole`, label it with the `servicebinding.io/controller` flag and set the flag value to `true`.
```

Author: @missmesss  Looks so much better now! Thanks :)

> The Operator service account is bound to an aggregated cluster role, allowing Operator providers or cluster administrators to enable binding custom service resources to workloads. You can enclose the required permissions within a `ClusterRole` by labeling it with the `servicebinding.io/controller` flag and setting the flag value to `true`.

Reviewer 2: The 2nd sentence (about ClusterRole and the flag) is probably not needed since it describes/repeats the procedure that is provided in the example below.
Also, it looks like the sentence doesn't describe the whole procedure and users will need to check the example anyway? Do I understand it correctly?

Author: @missmesss I like to retain as we need to tell them about the flag and setting its value. But I am removing the latter part of the lead sentence just before the example.

> The following example shows how to allow the {servicebinding-title} to `read`, `watch`, and `list` the custom resources (CRs) of Crunchy PostgreSQL Operator by enclosing the required permissions within a `ClusterRole` resource:

Reviewer 2: ```suggestion
The following example shows how to allow the {servicebinding-title} to `read`, `watch`, and `list` the custom resources (CRs) of the Crunchy PostgreSQL Operator:
```
The ending could possibly be removed since it describes what a user sees in the example, WDYT?

> The following example shows how to allow the {servicebinding-title} to `read`, `watch`, and `list` the custom resources (CRs) of Crunchy PostgreSQL Operator by enclosing the required permissions within a `ClusterRole` resource:

Reviewer 2: About `read`: The example includes `get`, can it be confusing to the users? It's clear that read=get, but maybe seeing 2 options can make users doubt which one to use?

Author: @missmesss It is `get`. My bad! Rectifying it. Thanks :)
